doc/platform_json_enhancement.md

```diff
@@ -0,0 +1,125 @@
+# Platform capability file enhancement #
+
+### Rev 0.1
+
+## Table of Content
+
+## Revision
+
+ | Rev |     Date    |            Author            | Change Description   |
+ |:---:|:-----------:|:----------------------------:|----------------------|
+ | 0.1 |             | Arun Saravanan Balachandran  | Initial version      |
+
+## Scope
+
+This document provides information on the enhancements for platform capability file `platform.json`.
+
+## Definitions/Abbreviations
+
+| Definitions/Abbreviation | Description |
+|--------------------------|-------------|
+| BMC | Baseband Management Controller |
+| NOS | Network Operating System |
+| PSU | Power Supply Unit |
+
+## Overview
+
+Each networking switch has a set of platform components (e.g: Fans, PSUs, LEDs, etc.) and these components in each platform can have different characteristics (like supported colors for a LED). In a given platform, the components could be controlled by a dedicated platform controller (like BMC) or the NOS running on the CPU is required to control it and in the former the control of the components from the NOS could be limited.
+
+In SONiC the platform components' supported attributes are made available via Platform API, but certain platform specific capabilties for the components are not available for the applications.
+
+This document provides the enhancement for `platform.json` to address the above issue.
+
+## Design
+
+Currently, `platform.json` is used for providing the expected structure of the platform components and interface details for supporting dynamic port breakout.
+
+### Platform capabilities field
+
+A new set of `capabilities` fields are introduced in platform.json, for providing platform specific capablities on control and characteristics of the components.
+
+For each component's attribute, the defined `capabilities` fields are as follows:
+
+- "control" : A boolean, 'true' if the given attribute can be controlled from the NOS, 'false' otherwise.
```

Comment:
Although it might be a bit longer, maybe this key should be named `controllable` for clarity?

Also, is there a default value assumed if this is omitted? I assume we would default to `true`?

Reply:
> Although it might be a bit longer, maybe this key should be named `controllable` for clarity?

As suggested, the name of the key has been changed from `control` to `controllable`

> Also, is there a default value assumed if this is omitted? I assume we would default to `true`?

For the default value, the application/test suite can assume it as `true` to maintain the current behavior

Reply:
> Could you add a note about this in the doc, please? If `controllable` is not specified, it is assumed to be true.

Have included the default value for 'controllable'